boards/components/src/apds9960.rs

```diff
@@ -2,48 +2,54 @@
 
 use capsules::apds9960::APDS9960;
 use capsules::virtual_i2c::{I2CDevice, MuxI2C};
+use core::marker::PhantomData;
 use core::mem::MaybeUninit;
 use kernel::component::Component;
 use kernel::hil::gpio;
+use kernel::hil::i2c;
 
 #[macro_export]
 macro_rules! apds9960_component_static {
-    () => {{
-        let i2c_device = kernel::static_buf!(capsules::virtual_i2c::I2CDevice<'static>);
+    ($I:ty $(,)?) => {{
+        let i2c_device = kernel::static_buf!(capsules::virtual_i2c::I2CDevice<'static, $I>);
         let apds9960 = kernel::static_buf!(capsules::apds9960::APDS9960<'static>);
         let buffer = kernel::static_buf!([u8; capsules::apds9960::BUF_LEN]);
 
         (i2c_device, apds9960, buffer)
     };};
 }
 
-pub struct Apds9960Component {
-    i2c_mux: &'static MuxI2C<'static>,
+pub struct Apds9960Component<I: 'static + i2c::I2CMaster, J: 'static + i2c::I2CDevice> {
+    i2c_mux: &'static MuxI2C<'static, I>,
     i2c_address: u8,
     interrupt_pin: &'static dyn gpio::InterruptPin<'static>,
+    i2c_device: PhantomData<J>,
 }
 
-impl Apds9960Component {
+impl<I: 'static + i2c::I2CMaster, J: 'static + i2c::I2CDevice> Apds9960Component<I, J> {
     pub fn new(
-        i2c_mux: &'static MuxI2C<'static>,
+        i2c_mux: &'static MuxI2C<'static, I>,
         i2c_address: u8,
         interrupt_pin: &'static dyn gpio::InterruptPin<'static>,
     ) -> Self {
         Apds9960Component {
             i2c_mux,
             i2c_address,
             interrupt_pin,
+            i2c_device: PhantomData,
         }
     }
 }
 
-impl Component for Apds9960Component {
+impl<I: 'static + i2c::I2CMaster, J: 'static + i2c::I2CDevice> Component
+    for Apds9960Component<I, J>
+{
     type StaticInput = (
-        &'static mut MaybeUninit<I2CDevice<'static>>,
-        &'static mut MaybeUninit<APDS9960<'static>>,
+        &'static mut MaybeUninit<I2CDevice<'static, I>>,
+        &'static mut MaybeUninit<APDS9960<'static, J>>,
```

Comment:
```suggestion
        &'static mut MaybeUninit<APDS9960<'static, I2CDevice<'static, I>>>,
```

This is the mistake, the component uses the `I2CDevice` structure.